### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by Yazino12**
### In this pull:

- Write `tests for all methods`
- Update `Readme.md`
- Initialize **Solver** class to see tests failing
- write methods for **Solver** class
- Add tests for `fizzbuzz` when values equal to 3, 15 and 7
- Add tests for `factorial` when value equal to 0

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      expect(solver.factorial(num)).to eq 120
+    end
+
+    it 'reverse of maano should be onaam' do
+      expect(solver.reverse(word)).to match(/^onaam+$/)
+    end
+
+    it 'fizzbuzz should return buzz for value 5' do
+      expect(solver.fizzbuzz(num)).to eq 'buzz'
+    end
```
Comment (AshakaE) [inline spec/solver_spec.rb:27]: - [x] Kindly write tests to handle
- `fizz` (divisble by 3)
- `fizzbuzz`(divisble by 5)
- a number not divible by 3 or 5

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      expect(solver).to respond_to(:factorial)
+    end
+
+    it 'factorial of 5 should be 120' do
+      expect(solver.factorial(num)).to eq 120
+    end
+
+    it 'reverse of maano should be onaam' do
+      expect(solver.reverse(word)).to match(/^onaam+$/)
+    end
```
Comment (AshakaE) [inline spec/solver_spec.rb]: - [x] kindly write a factorial test that handles a special case when the input is 0

**Review by AshakaE — CHANGES_REQUESTED** [review]
Hi @Yazino12 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Git flow ✅️


## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 2 above)
Comment (Yazino12) [reply spec/solver_spec.rb]: I will implement these changes as well 🚀.

File: spec/solver_spec.rb (hunk 1 above)
Comment (Yazino12) [reply spec/solver_spec.rb:27]: @AshakaE, Thanks for taking the time 😇, I will implement these changes right away 🚀.

**Review by danielufeli — APPROVED** [review]
Hi @Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

a

## Highlights
- No Linter Errors✔️
- Clean Code ✔️
- Professional Repo✔️
- Descriptive Readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @danielufeli in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**